src/nodes.rs

```diff
@@ -0,0 +1,106 @@
+use crate::geoip_cache::GeoIPCache;
+use ziggurat_core_geoip::geoip::GeoInfo;
+
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+use spectre::graph::{AGraph, Graph};
+
+#[derive(Default, Clone, Serialize, Deserialize)]
+pub struct Node {
+    // the ip address, dotted quad, without port number
+    pub ip: String,
```

Comment:
```suggestion
    /// the ip address, dotted quad, without port number
    pub ip: String,
```
in Rust, `///` are used for commenting public items (should be applied for all elements here)

Reply:
done